### ISSUE TITLE
spring-boot-maven-plugin forks JVM which keeps running after failing integration tests

### ISSUE THREAD

**Issue by smurf667** [spring-projects/spring-boot#18948]
With Spring Boot 2.2.0 the default for forking was changed from `false` to `true` (see https://github.com/spring-projects/spring-boot/issues/18638).

https://docs.spring.io/spring-boot/docs/current/maven-plugin/start-mojo.html suggests to start the application in `pre-integration-test` and to stop it in `post-integration-test`.

When an integration test fails and the application JVM was forked, it will continue to run. That is sub-optimal. Maybe the example should suggest _not_ to fork?

Issue demonstration at https://github.com/smurf667/spring-boot-fork-demo

**Comment by wilkinsona** [comment]
Thanks for the sample. Things appear to be working as designed. In the `post-integration-test` phase, I can see that a request is made to stop the application:

```
[INFO] --- spring-boot-maven-plugin:2.2.1.RELEASE:stop (post-integration-test) @ demo ---
[INFO] Stopping application...
2019-11-09 09:51:26.243  INFO 21928 --- [on(5)-127.0.0.1] inMXBeanRegistrar$SpringApplicationAdmin : Application shutdown requested.
```

The application JVM will then exit once all non-daemon threads have stopped. Normally, this would happen as a result of the application context being closed. It does not happen in your sample as the timer task creates a non-daemon thread that does not stop when the context is closed.

If you update your sample to depend on `spring-boot-starter-web` (which starts some threads for handling web requests) and remote the use of a timer task, the shutdown completes in a timely manner as the HTTP request handling threads stop as part of the application context being closed:

```
[INFO] --- spring-boot-maven-plugin:2.2.1.RELEASE:stop (post-integration-test) @ demo ---
[INFO] Stopping application...
2019-11-09 09:55:21.524  INFO 22916 --- [on(2)-127.0.0.1] inMXBeanRegistrar$SpringApplicationAdmin : Application shutdown requested.
2019-11-09 09:55:21.527  INFO 22916 --- [on(2)-127.0.0.1] o.s.s.concurrent.ThreadPoolTaskExecutor  : Shutting down ExecutorService 'applicationTaskExecutor'
```

In short, you need to ensure that the lifecycle of any non-daemon threads that are started by your application are tied to the application context. This ensures that the threads will stop when the context is closed, allowing the JVM to shut down when requested.
